ipa-core/src/test_fixture/mod.rs

```diff
@@ -147,6 +147,37 @@ where
     join3(fut0, fut1, fut2)
 }
 
+/// Wrapper for joining the first item from three iterator in an iterator into an array.
+/// # Panics
+/// If the tasks return `Err` or if `a` is the wrong length.
+pub fn join3v0<T, I, V>(a: V) -> JoinAll<T>
```

Comment:
```suggestion
pub fn join3v0<T, I, V>(a: V) -> impl Future<Output = ?> 
```

Reply:
also a better name would be helpful here - I suspect that this may get used quite a bit, so `join3flatten` maybe?

Reply:
good call out. the comment needs updating too. the original intention was to have it be like `join3v` but only return the result from shard 0, but then the functionality shifted a bit as I was working out how to get everything to actually poll.